### PR TITLE
CSV functionality

### LINKED ISSUE #106: Add donations through a CSV file
- Allow user to add donations by import CSV 
- We need to filter through the CSV and save the info (columns) we want 
- Call the create a donation #56 endpoint repeatedly to add all the info to our database

- Add button to the donor's page that will call this endpoint

image

### PR DESCRIPTION
**Pull request by CurtisMIT**
# Summary

Write a quick summary of your changes here. Make sure to provide context for people that may be unfamiliar with what you're working on!

Route and unit test for receiving and parsing csvs into `Donation`s. Go through the uploaded csv files, for each csv file, for each row in that csv file, construct a `Donation`. Both `event` and `num_tickets` are optional fields. Still waiting on @celinehuang 's input on whether `added_by` should be optional. I personally think it shouldn't be as you have to be a logged in user to click the upload csv button.

Edit: @celinehuang approved of the `added_by` change from optional to must

## Test Plan

How did you test your changes?
- Passed all unit tests

## Related Issues

Which issues does this PR resolve/work on?
Closes #106

### CODE REVIEW

**Review by peterghrong — COMMENTED** [review]
NICE

**Comment by celinehuang** [comment]
are you hoping to do the frontend in another PR? or would be make sense to add the button on the frontend placeholder page as well

**Comment by CurtisMIT** [comment]
@celinehuang yea I'm hoping to do it in another pr, or do you want me to do it in the same pr?

File: backend/app/donation/routes.py [hunk 1]
```diff
@@ -114,3 +128,64 @@ def get_donation_amount():
 
         total_amount = sum([donation.amount for donation in filtered_donation])
         return jsonify(total_amount=total_amount)
+
+
+@donation.route("/csvs", methods=["POST"])
```
Comment (celinehuang) [inline backend/app/donation/routes.py:133]: how does `/donation-csv` sound? I have a feeling we might have to do something similar for contacts especially when we want to populate prod db and the fields would be different

Comment (CurtisMIT) [reply backend/app/donation/routes.py]: oh didn't know we were doing it for contacts as well, though this route is prefixed with `/donation` anyways as it is part of the `donation` blueprint. Should I still change it to `donation-csv`?

Comment (celinehuang) [reply backend/app/donation/routes.py:133]: oh right I totally forgot that's how Flask work

Comment (celinehuang) [reply backend/app/donation/routes.py:133]: csv is probs good? no strong opinions though csvs work too

Comment (CurtisMIT) [reply backend/app/donation/routes.py]: done

**Review by celinehuang — APPROVED** [review]
Looking forward to the frontend PR ⭐

**Pull request merged**